common/edl/fcntl.edl

```diff
@@ -15,9 +15,40 @@
 enclave
 {
 
-    include "openenclave/internal/syscall/dirent.h"
-    include "openenclave/internal/syscall/sys/stat.h"
-    include "openenclave/internal/syscall/types.h"
+    include "openenclave/bits/edl/syscall_types.h"
+
+    struct oe_dirent
+    {
+        uint64_t d_ino;
+        oe_off_t d_off;
+        uint16_t d_reclen;
+        uint8_t d_type;
+        char d_name[256]; // OE_NAME_MAX + 1 == 256. TODO: Math in EDL?
```

Comment:
I assume deep copy should be able to handle fixed size array?

Reply:
I think just normal shallow copy will handle it. Deep copy is just for following pointers.

Reply:
Do you mean that this array does not need to be deep copied? I'm just thinking whether deep copy can infer the size of the array.

Reply:
I mean normal EDL can get the size of the array. This will still get copied even if the deep copy feature is not enabled

Reply:
Got it! Thanks for the clarification.